### PR TITLE
Remove all conditionals and references to Android 4.x

### PR DESCRIPTION
**Pull request by marcoconti83**
### What's new in this PR?

Remove `if(Android < 5)` conditions and cleaned up API level constants. These were leftovers from removing support from Android 4.x and lower. 

We now assume an API level >= 21 everywhere.

### Testing

I tested the build on an emulator with API version 22 and it worked fine
#### APK
[Download build #85](http://10.10.124.11:8080/job/Pull%20Request%20Builder/85/artifact/build/artifact/wire-dev-PR2317-85.apk)
[Download build #87](http://10.10.124.11:8080/job/Pull%20Request%20Builder/87/artifact/build/artifact/wire-dev-PR2317-87.apk)
[Download build #88](http://10.10.124.11:8080/job/Pull%20Request%20Builder/88/artifact/build/artifact/wire-dev-PR2317-88.apk)
[Download build #89](http://10.10.124.11:8080/job/Pull%20Request%20Builder/89/artifact/build/artifact/wire-dev-PR2317-89.apk)

### CODE REVIEW

**Comment by marcoconti83** [comment]
build again

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/bitmap/video/VideoTranscoder.scala [hunk 1]
```diff
@@ -66,8 +66,7 @@ object VideoTranscoder {
   }
 
   def apply(context: Context): VideoTranscoder =
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) new VideoTranscoder18(context)
-    else new FallbackTranscoder(context)
```
Comment (johnxnguyen) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/bitmap/video/VideoTranscoder.scala]: Looks like this is the only place used for `FallbackTranscoder`. We could safely delete that now.

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/service/ContactsService.scala [hunk 2]
```diff
@@ -466,7 +465,7 @@ object ContactsServiceImpl {
     val InDefaultDirectory = ContactsContract.ContactsColumns.IN_DEFAULT_DIRECTORY
   }
 
-  lazy val Visible = if (SDK_INT >= LOLLIPOP) Some(s"${Col.Visible} = 1 OR ${Col.InDefaultDirectory} = 1") else None
+  lazy val Visible = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) Some(s"${Col.Visible} = 1 OR ${Col.InDefaultDirectory} = 1") else None
```
Comment (johnxnguyen) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/service/ContactsService.scala]: Do we need this conditional? The all builds should now be >= `LOLLIPOP`.

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/service/NetworkModeService.scala [hunk 3]
```diff
@@ -73,7 +74,7 @@ class DefaultNetworkModeService(context: Context, lifeCycle: UiLifeCycle) extend
 
   //this method doesn't really belong here, but it's only used for tracking - just tells us if the device was in doze mode or not
   //fixme get the version code from android sdk
-  @TargetApi(23)
+  @TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR2)
```
Comment (johnxnguyen) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/service/NetworkModeService.scala]: `JELLY_BEAN_MR2` is API level 18. I think you mean `M` here.

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/utils/Locales.scala [hunk 4]
```diff
@@ -44,7 +43,7 @@ object Locales extends DerivedLogTag {
       locale    <- localeOptFromConfig(config)
     } yield locale) getOrElse Locale.getDefault
 
-  lazy val bcp47 = if (SDK_INT >= LOLLIPOP) AndroidLanguageTags.create else FallbackLanguageTags.create
+  lazy val bcp47 = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) AndroidLanguageTags.create else FallbackLanguageTags.create
```
Comment (johnxnguyen) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/utils/Locales.scala]: Again, I don't think you need this conditional, which also means you can delete `FallbackLanguageTags`

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/service/ContactsService.scala (hunk 2 above)
Comment (marcoconti83) [reply wire-android-sync-engine/zmessaging/src/main/scala/com/waz/service/ContactsService.scala]: Correct!

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/service/NetworkModeService.scala (hunk 3 above)
Comment (marcoconti83) [reply wire-android-sync-engine/zmessaging/src/main/scala/com/waz/service/NetworkModeService.scala]: Well spotted :)

**Pull request merged**